js/modules/k6/grpc/client_test.go

```diff
@@ -69,16 +71,558 @@ func assertMetricEmitted(t *testing.T, metric *stats.Metric, sampleContainers []
 	assert.True(t, seenMetric, "url %s didn't emit %s", url, metric.Name)
 }
 
+// Run represents an execution of a k6 script.
+type codeBlock struct {
+	code    string
+	val     interface{}
+	err     string
+	asserts func(*testing.T, *httpmultibin.HTTPMultiBin, chan stats.SampleContainer)
+}
+
+type testcase struct {
+	name       string
+	setup      func(*httpmultibin.HTTPMultiBin)
+	initString codeBlock // runs in the init context
+	vuString   codeBlock // runs in the vu context
+}
+
 func TestClient(t *testing.T) {
 	t.Parallel()
-	tb := httpmultibin.NewHTTPMultiBin(t)
-	sr := tb.Replacer.Replace
+	tests := []testcase{
+		{
+			name: "New",
+			initString: codeBlock{
+				code: `
+			var client = new grpc.Client();
+			if (!client) throw new Error("no client created")`,
+			},
+		},
+		{
+			name: "LoadNotFound",
+			initString: codeBlock{
+				code: `
+			var client = new grpc.Client();
+			client.load([], "./does_not_exist.proto");`,
+				// (rogchap) this is a bit of a hack as windows reports a different system error than unix
+				err: "no such file or directory|The system cannot find the file specified",
```

Comment:
I'd make this explicit with an `if runtime.GOOS == "windows" { ... } else { ... }` check instead. And we can avoid the regex ;)